src/receiver-ui.py

```diff
@@ -68,13 +68,20 @@ def receive():
 def check_status():
     stream_ids = list(receiver.processes.keys())
     for stream_id in stream_ids:
-        stream_deleted = True
-        for stream in receiver.streams:
-            if stream["id"] == int(stream_id):
-                stream_deleted = False
-        if stream_deleted:
+        # First, check if stream has been manually closed
+        if receiver.processes[stream_id][0].poll() is not None:
             for process in receiver.processes[stream_id]:
                 process.terminate()
+            del receiver.processes[stream_id]
+            receiver.delete_stream(stream_id)
+        else:
+            stream_deleted = True
+            for stream in receiver.streams:
+                if stream["id"] == int(stream_id):
+                    stream_deleted = False
+            if stream_deleted:
+                for process in receiver.processes[stream_id]:
+                    process.terminate()
                 del receiver.processes[stream_id]
```

Comment:
There's a little bit of code duplication with this code, the one in the first `if` of the method, and the analogous method in sender-ui. it could be nice to extract it.

Reply:
Good point - I'll make that change in my next PR that's coming